### DISCUSSION TITLE
No matching primary target from server

### DISCUSSION THREAD

**Discussion by dappa** [unpoly/unpoly#357] [Q&A] [answered]
I'm having some problems with updating a fragment from the server when submitting a form.

Example: https://jsfiddle.net/y7b9tcd6/50/

```html
<table>
   <thead>
       <tr>
           <th></th>
           <th></th>
       </tr>
   </thead>
   <tfoot>
       <tr>
          <td>
              <form action="/echo/html/" id="form" method="post" up-submit up-target="tr"></form>
              Add new:
          </td>
          <td><button form="form">Submit</button></td>
       </tr>
   </tfoot>
</table>
```

Server response
```html
<tr>
    <td>Changed tfoot tr</td>
</tr>
```

The problem is that I get the error `Could not match primary target (tr). Updating a fallback target (body).`
I thought the `up-target` would scan for the `closest` matching element from the `form` tag and then swap with the matching tag from the server. Anyone know what I'm doing wrong here?

**Comment by dappa** [comment]
The documentation does say that

`The one exception is that when a DOM element is passed as { target } or [{ origin }](https://unpoly.com/origin) option, Unpoly will use that element's layer. Clicking an Unpoly-enabled [link](https://unpoly.com/a-up-follow) or [form](https://unpoly.com/form-up-submit) will automatically set that link or form as the [{ origin }](https://unpoly.com/origin).`

Am I correct in assuming that it only tries to find `tr` tag within the `origin` layer, which in my case is just an empty form tag?

**Comment by adam12** [comment, marked as answer]
You'll need to return a larger fragment. Just returning a `<tr>` is not enough.

```
<input type="hidden" name="html" value="<table><tfoot><tr><td>CONTENT</td></tr></tfoot></table>">
```

This might work for other elements, but IIRC `<table>` elements and their children react differently.

Using the `:content` pseudo selector _might_ work, but it raises an exception for me. I've never used it before so perhaps Henning could comment on that further.

**Reply by dappa** [reply]
Thanks @adam12 , using a `table` did the trick!
